### PR TITLE
RGRIDT 849 - Defining column dependency - Other column types

### PR DESCRIPTION
**Pull request by JoaoFerreira-FrontEnd**
### What was happening
*  Wasn't possible to bind an interaction of a cell with the content of another one in the same row.

### What was done
* A new method SetCellData() was created;

### Test Steps
1. Add an action on OnCellValueChange available on the column
2. Inside this action use the SetCellData client action to change a value of another cell inside this row.
3. Check if the second cell is edited, after the first one is edited. 

### Checklist
* [x] tested locally
* [x] documented the code
* [x] clean all warnings and errors of eslint
* [x] requires changes in OutSystems
* [x] requires new sample page in OutSystems

### CODE REVIEW

File: code/src/GridAPI/Cells.ts [hunk 1]
```diff
@@ -40,4 +40,34 @@ namespace GridAPI.Cells {
...
+        gridID: string,
+        rowIndex: number,
+        columnID: string,
+        // eslint-disable-next-line
+        value: any,
+        showDirtyMark = true
+    ): void {
+        if (OSFramework.Helper.IsGridReady(gridID) === false) return;
+        const grid = GridManager.GetGridById(gridID);
+        const column = ColumnManager.GetColumnById(columnID).provider;
```
Comment (tmbp95) [inline code/src/GridAPI/Cells.ts:63]: We should avoid using the provider here and inside any other API method

File: code/src/GridAPI/ContextMenu.ts [hunk 2]
```diff
@@ -17,8 +17,9 @@ namespace GridAPI.ContextMenu {
     ): string {
         //Try to find in DOM only if not present on Map
         if (lookUpDOM && !_menuItemsToGridId.has(menuItemId)) {
-            const menuOptionElement =
-                OSFramework.Helper.GetElementByUniqueId(menuItemId);
+            const menuOptionElement = OSFramework.Helper.GetElementByUniqueId(
```
Comment (tmbp95) [inline code/src/GridAPI/ContextMenu.ts:20]: Please run lint here and auto-fix this error

File: code/src/OSFramework/Interface/IDataSource.ts [hunk 3]
```diff
@@ -69,5 +69,10 @@ namespace OSFramework.Grid {
          * @param dataItem
          */
         toOSFormat(dataItem: any): any;
+        /**
+         * Transform value to JSON format
```
Comment (tmbp95) [inline code/src/OSFramework/Interface/IDataSource.ts:73]: By the name of the method I would not think that the method was going to transform value to JSON format :p

File: code/src/WijmoProvider/Features/CellData.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        public build(): void {
+            //
+        }
+
+        public setCellData(
+            rowNumber: number,
+            columnID: string,
+            value: string
+        ): void {
+            const column = GridAPI.ColumnManager.GetColumnById(columnID);
```
Comment (tmbp95) [inline code/src/WijmoProvider/Features/CellData.ts:25]: We should avoid doing API calls inside the provider

File: code/src/WijmoProvider/Helper/DataUtils.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        if (hasTimeZone) {
+            ticks = value.getTime();
+        } else {
+            ticks = value.getTime() - value.getTimezoneOffset() * 60 * 1000;
+        }
+
+        return ticks;
+    }
+
+    export function TrimSeconds(value: string): string {
```
Comment (tmbp95) [inline code/src/WijmoProvider/Helper/DataUtils.ts:18]: What is the difference between this method and the "trimSecondsFromDate" previously defined?

File: code/src/OSFramework/Interface/IDataSource.ts (hunk 3 above)
Comment (bmartinho) [reply code/src/OSFramework/Interface/IDataSource.ts:73]: Aren't you looking at the description of the next method?

File: code/src/WijmoProvider/Helper/DataUtils.ts (hunk 5 above)
Comment (bmartinho) [reply code/src/WijmoProvider/Helper/DataUtils.ts:18]: The TrimSecondsForDate will trim the seconds: 2021-06-25 12:23:34 => 2021-06-25 12:23 -> This is required format for SetCellData from wijmo

The TrimSeconds will clean the seconds: 2021-06-25T12:23:34Z => 2021-06-25T12:23:00Z -> This is required for comparisons in Conditional Format - Will rename this!

File: code/src/OSFramework/Feature/ICellData.ts [hunk 6]
```diff
@@ -10,7 +10,7 @@ namespace OSFramework.Feature {
          */
         setCellData(
             rowNumber: number,
-            columnID: string,
+            columnID: OSFramework.Column.IColumn,
```
Comment (tmbp95) [inline code/src/OSFramework/Feature/ICellData.ts:13]: can you please change this to column instead of columnID

**Review by bmartinho — APPROVED** [review]
Ready to merge

**Review by tmbp95 — APPROVED** [review]

**Pull request merged**